File: code-samples/messaging/code-snippets/send-mms.java
package Send_MMS;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class Send_MMS {
    static String RECIPIENT = "RECIPIENT-PHONE-NUMBER";
    static RestClient restClient;

    public static void main(String[] args) {
      // Instantiate the SDK
      restClient = new RestClient( "SANDBOX-APP-CLIENTID", "SANDBOX-APP-CLIENTSECRET", "https://platform.devtest.ringcentral.com");
      var obj = new Send_MMS();
      try {
        // Authenticate a user using a personal JWT token
        restClient.authorize( "SANDBOX_JWT" );
        obj.read_extension_phone_number_detect_mms_feature();
      } catch (RestException | IOException e) {
        e.printStackTrace();
      }
    }
    /*
    Read phone number(s) that belongs to the authenticated user and detect if a phone number
    has the MMS capability
    */
    public void read_extension_phone_number_detect_mms_feature() throws IOException{
      try {
        var resp =  restClient.restapi().account().extension().phoneNumber().get();
        for (var record : resp.records) {
          if (record.usageType.equalsIgnoreCase("DirectNumber")) {
            for(var feature : record.features) {
              if (feature.equalsIgnoreCase("MmsSender")) {
                // If a user has multiple phone numbers, check and decide which number
                // to be used for sending MMS message.
                send_mms(record.phoneNumber);
                return;
              }
            }
          }
        }
        if (resp.records.length == 0) {
          System.out.println("This user does not own a phone number!");
        } else {
          System.out.println("None of this user's phone number(s) has the MMS capability!");
        }
      }catch(RestException e) {
        System.out.println(e.getMessage());
      }
    }
    /*
    Send a multi-media message from a user own phone number to a recipient number
    */
    public void send_mms(String fromNumber) throws RestException, IOException{
      try {
        CreateMMSMessage bodyParams = new CreateMMSMessage();
        bodyParams.from = new MessageStoreCallerInfoRequest().phoneNumber(fromNumber);

        bodyParams.to = new MessageStoreCallerInfoRequest[]{
          new MessageStoreCallerInfoRequest().phoneNumber(RECIPIENT)
        };
        // To send group messaging, add more (max 10 recipients) 'phoneNumber' object. E.g.
        /*
        requestBody.to = new MessageStoreCallerInfoRequest[] {
          new MessageStoreCallerInfoRequest().phoneNumber("12092520012"),
          new MessageStoreCallerInfoRequest().phoneNumber("16505130930")
        };
        */

        Attachment attachment = new Attachment()
            .filename("test.jpg")
            .content(Files.readAllBytes(Paths.get("./src/test/resources/test.jpg")));
        Attachment[] attachments = new Attachment[] { attachment };
        bodyParams.attachments = attachments;

        var response = restClient.restapi().account().extension().mms().post(bodyParams);
        System.out.println("MMS sent. Message id: " + response.id.toString());
        check_mms_message_status(response.id.toString());
      } catch(RestException e) {
        System.out.println(e.getMessage());
      }
    }
    /*
    Check the sending message status until it's no longer in the queued status
    */
    private void check_mms_message_status(String messageId) throws IOException {
      try {
        var resp = restClient.restapi().account().extension().messageStore(messageId).get();
        System.out.println("Message status: " + resp.messageStatus);
        if (resp.messageStatus.equals("Queued")) {
          try {
            Thread.sleep(5000);
            check_mms_message_status(resp.id.toString());
          } catch (InterruptedException e) {
            e.printStackTrace();
            System.out.println(e);
          }
        }
      } catch (RestException e) {
        System.out.println(e.getMessage());
      }
    }
}

File: code-samples/messaging/code-snippets/send-mms.cs
using System;
using System.Threading.Tasks;
using RingCentral;

namespace Send_MMS
{
    class Program
    {
        static RestClient restClient;
        static string RECIPIENT = "RECIPIENT-PHONE-NUMBER";
        static async Task Main(string[] args)
        {
          try
          {
            // Instantiate the SDK
            restClient = new RestClient( "RC_APP_CLIENT_ID", "RC_APP_CLIENT_SECRET", "https://platform.ringcentral.com");
            // Authenticate a user using a personal JWT token
            await restClient.Authorize("RC_USER_JWT");

            await read_extension_phone_number_detect_mms_feature();
          }
          catch (Exception ex)
          {
            Console.WriteLine(ex.Message);
          }
        }

        /*
          Read phone number(s) that belongs to this user and detect if a phone number
          has the MMS capability
        */
        static private async Task read_extension_phone_number_detect_mms_feature()
        {
            try
            {
                var resp = await restClient.Restapi().Account().Extension().PhoneNumber().Get();
                foreach (var record in resp.records)
                {
                    foreach (var feature in record.features)
                    {
                        if (feature == "MmsSender")
                        {
                            // If a user has multiple phone numbers, check and decide which number
                            // to be used for sending MMS message.
                            await send_mms(record.phoneNumber);
                            return;
                        }
                    }
                }
                if (resp.records.Length == 0)
                {
                    Console.WriteLine("This user does not own a phone number!");
                }
                else
                {
                    Console.WriteLine("None of this user's phone number(s) has the MMS capability!");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
        /*
         Send a multi-media message from a user own phone number to a recipient number
        */
        static private async Task send_mms(string fromNumber)
        {
            try
            {
                var bodyParams = new CreateMMSMessage();
                bodyParams.from = new MessageStoreCallerInfoRequest
                {
                    phoneNumber = fromNumber
                };
                bodyParams.to = new MessageStoreCallerInfoRequest[] {
                    new MessageStoreCallerInfoRequest { phoneNumber = RECIPIENT }
                };
                // To send group messaging, add more (max 10 recipients) 'phoneNumber' object. E.g.
                /*
                requestBody.to = new MessageStoreCallerInfoRequest[] {
                  new MessageStoreCallerInfoRequest { phoneNumber = "Recipient_1_Number" },
                  new MessageStoreCallerInfoRequest { phoneNumber = "Recipient_2_Number" }
                };
                */
                bodyParams.text = "Hello World!";

                var attachment = new Attachment { filename = "test.jpg", contentType = "image/jpeg", content = System.IO.File.ReadAllBytes("test.jpg") };
                var attachments = new Attachment[] { attachment };
                bodyParams.attachments = attachments;

                var resp = await restClient.Restapi().Account().Extension().Mms().Post(bodyParams);
                Console.WriteLine("MMS sent. Message id: " + resp.id.ToString());
                await check_mms_message_status(resp.id.ToString());
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
        /*
         Check the sending message status until it's no longer in the queued status
        */
        static private async Task check_mms_message_status(string messageId)
        {
            try
            {
                var resp = await restClient.Restapi().Account().Extension().MessageStore(messageId).Get();
                Console.WriteLine("Message status: " + resp.messageStatus);
                if (resp.messageStatus == "Queued")
                {
                    Thread.Sleep(2000);
                    await check_mms_message_status(resp.id.ToString());
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
    }
}
